### PR TITLE
build: remove some unused COCKROACH_PROPOSER_EVALUATED_KV vars

### PR DESCRIPTION
**Pull request by dt**

### CODE REVIEW

**Comment by tamird** [comment]
:lgtm: but let's wait until after this week's beta to merge.

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15183#-:-Ki6vuxHtignN-GBN2HL:bhgyamr)*

**Review by bdarnell — APPROVED** [review]

**Comment by dt** [comment]
... good to go here?

**Comment by bdarnell** [comment]
LGTM

**Pull request merged**
